### PR TITLE
[Feature][Quantization] auto_round support for mixed bits quantization

### PR DESCRIPTION
**Pull request by n1ck-guo**
## Purpose
Add support for mixed bits quantization in autoround config

## Test Plan
Load auto_round quantized model with extra_config(including layers fallback to 16 bits).

## Test Result
With the change, per-layer excludes (e.g., model.layers.0.*) are respected.
Unquantized layers return UnquantizedLinearMethod.
Successful load mixed bits quantization model with auto-round quant_method

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/layers/quantization/auto_round.py [hunk 1]
```diff
@@ -378,6 +378,11 @@ def apply_ipex_quant_layer(self, layer, prefix: str):
             return None
 
     def get_quant_method(self, layer: torch.nn.Module, prefix: str):
+        if prefix:
+            for layer_name in self.extra_config:
+                if layer_name == prefix or layer_name.endswith(prefix):
```
Comment (mgoin) [inline vllm/model_executor/layers/quantization/auto_round.py]: Should the latter condition be flipped? If prefix ends with layer name? Usually prefix is the absolute name in the state dict

File: vllm/model_executor/layers/quantization/auto_round.py [hunk 2]
```diff
@@ -378,6 +378,11 @@ def apply_ipex_quant_layer(self, layer, prefix: str):
...
+                    if self.extra_config[layer_name]['bits'] >= 16:
```
Comment (mgoin) [inline vllm/model_executor/layers/quantization/auto_round.py]: Is bits guaranteed for each layer? Maybe it should be a get that defaults to 16?

**Review by mgoin — COMMENTED** [review]
LGTM thanks for the PR. Just some comments and make sure to fix the precommit

File: vllm/model_executor/layers/quantization/auto_round.py (hunk 2 above)
Comment (n1ck-guo) [reply vllm/model_executor/layers/quantization/auto_round.py]: If layer_name in the extra_config, it must have bits. However, i will add a default value.

File: vllm/model_executor/layers/quantization/auto_round.py (hunk 1 above)
Comment (n1ck-guo) [reply vllm/model_executor/layers/quantization/auto_round.py]: I debugged and found that the passed in prefix is not always the absolute name in the state dict. For example, the name in static dict is "model.layers.8.self_attn.k_proj". I found that prefix may be layers.8.self_attn.k_proj. I can modify the judgment method like layer_name == prefix or layer_name == f"model.{layer_name}"

**Comment by n1ck-guo** [comment]
@mgoin @tlrmchlsmth @yewentao256 @robertgshaw2-redhat  could you please help to review this.

**Review by yewentao256 — COMMENTED** [review]
Looks reasonable to me, could you also add a `lm_eval` report to make sure we don't hurt the accuracy?

**Review by yewentao256 — APPROVED** [review]
LGTM, please also fix the pre-commit issue.
Thanks for the work!

**Review by yewentao256 — APPROVED** [review]
